File: LeetCode/66.Plus_One/main.cpp
#include "solution.h"

// Unit Test

int main(){
    Solution s;
    vector<int> v1 {1,2,3};
    vector<int> v2 {4,3,2,1};
    vector<int> v3 {9,9,9};
    
    vector<int> res1 = s.plusOne(v1);
    vector<int> res2 = s.plusOne(v2);
    vector<int> res3 = s.plusOne(v3);


    for (int i=0; i!=v1.size(); i++){
        if (v1[i]!=res1[i]){ cout << "Failed 1 - v1" << endl; }
    }
    for (int i=0; i!=v2.size(); i++){
        if (v2[i]!=res2[i]){ cout << "Failed 2 - v2" << endl; }
    } 
    for (int i=0; i!=v3.size(); i++){
        if (v3[i]!=res3[i]){ cout << "Failed 3 - v3" << endl; }
    }

    return 0;
}

File: LeetCode/66.Plus_One/main.py
#!/usr/bin/python

from solution import Solution

# Unit Test
string1 = [1,2,3]
string2 = [4,3,2,1]
string3 = [9,9]
s = Solution()
if (s.plusOne(string1)==[1,2,4]):
    print("Pass")
else:
    print("Test 1 Failed")
if (s.plusOne(string2)==[4,3,2,2]):
    print("Pass")
else:
    print("Test 2 Failed")
if (s.plusOne(string3)==[1,0,0]):
    print("Pass")
else:
    print("Test 3 Failed")